lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -825,7 +831,17 @@ namespace Js
                 childrenModuleSet->Map([](LPCOLESTR specifier, SourceTextModuleRecord* moduleRecord)
                 {
                     Assert(moduleRecord->WasParsed());
-                    moduleRecord->ModuleDeclarationInstantiation();
+                    moduleRecord->shouldGenerateRootFunction =
+                            moduleRecord->ModuleDeclarationInstantiation();
+                });
+
+                childrenModuleSet->Map([](LPCOLESTR specifier, SourceTextModuleRecord* moduleRecord)
+                {
+                    if (moduleRecord->shouldGenerateRootFunction)
+                    {
+                        moduleRecord->shouldGenerateRootFunction = false;
+                        moduleRecord->GenerateRootFunction();
```

Comment:
Can GenerateRootFunction set the above flag false itself?

Reply:
It's actually false rest of the times.